### PR TITLE
Break out runtime lib to main and worker

### PR DESCRIPTION
**Pull request by kitsonk**
This PR splits out the runtime type library between a main and a web worker type library.  This makes it possible for the compiler to support compilation requests the are then validated against a more accurate type library.

cc/ @bartlomieju this should be 90% of it... requests to the compiler should include a `"type"` option with a value of `"main" | "worker" | "runtime"` and they will be used when compiling TypeScript.  It messes a bit with `deno types` as now there are two files that need to be concatenated together and there will be a triple-slash reference to `deno` lib, which I think ideally we would want to strip out of the file to make sure that `deno types` emits something that is valid.

### CODE REVIEW

**Comment by bartlomieju** [comment]
Just a heads up: this PR adds 2Mb to the compiler snapshot and thus to the executable. This is probably related to the fact that `lib.deno_main.d.ts` and `lib.deno_runtime.d.ts` are now included (as well as `lib.webworker.d.ts`).

File: cli/js/compiler_host.ts [hunk 1]
```diff
@@ -6,8 +6,20 @@ import { cwd } from "./dir.ts";
...
+ * compiler of what types should be used to validate the program against. */
+export enum CompilerHostTarget {
+  /** The main isolate library, where the main program runs. */
+  Main = "main",
+  /** The runtime API library. */
+  Runtime = "runtime",
+  /** The worker isolate library, where worker programs run. */
+  Worker = "worker"
+}
+
```
Comment (bartlomieju) [inline cli/js/compiler_host.ts]: So...
- `Runtime` includes Deno runtime declaration
- `Main` includes Deno runtime declaration as well as Window declaration.
- `Worker` includes Deno runtime and Worker declaration

If this is correct, that's just perfect for problem at hand!

Comment (kitsonk) [reply cli/js/compiler_host.ts]: At the moment, runtime and main are the same, in that they both include Window. I suspect in the future we may want to vary that. Your statement the rest is valid.

**Comment by kitsonk** [comment]
I do think it is temporary. If we could elude, or read on demand the files from deno_typescript, we could reduce the bundle size, because the compiler  bootstrap reads them into the isolate and they become part of the snapshot on a way they aren't needed anymore.

**Pull request merged**